src/neo4j_genai/components/types.py

```diff
@@ -100,13 +101,13 @@ class Neo4jRelationship(BaseModel):
     embedding_properties: Optional[list[Neo4jEmbeddingProperty]] = None
 
 
-class Neo4jGraph(BaseModel):
+class Neo4jGraph(DataModel):
     """Represents a Neo4j graph.
 
     Attributes:
         nodes (list[Neo4jNode]): A list of nodes in the graph.
         relationships (list[Neo4jRelationship]): A list of relationships in the graph.
     """
 
-    nodes: list[Neo4jNode]
-    relationships: list[Neo4jRelationship]
+    nodes: list[Neo4jNode] = []
+    relationships: list[Neo4jRelationship] = []
```

Comment:
Do we need defaults for these attributes? I think there will be an issue as these are mutable defaults

Reply:
Sorry looks like this is fine in Pydantic https://docs.pydantic.dev/latest/concepts/models/#fields-with-non-hashable-default-values

Reply:
It's useful to create the objects when there is a failure. And yes pydantic handles the mutable default properly, one of the magic parts of pydantic :)